remme/tp/node_account.py

```diff
@@ -0,0 +1,103 @@
+import logging
+from sawtooth_sdk.processor.exceptions import InvalidTransaction
+
+from remme.protos.node_account_pb2 import (
+    NodeAccount,
+    NodeAccountMethod,
+    NodeAccountInternalTransferPayload,
+    CloseMasternodePayload
+)
+
+from remme.shared.forms import (
+    NodeAccountInternalTransferPayloadForm,
+    CloseMasternodePayloadForm
+)
+
+from remme.settings import SETTINGS_MINIMUM_STAKE
+
+from .basic import (
+    PB_CLASS,
+    PROCESSOR,
+    VALIDATOR,
+    BasicHandler,
+    get_data,
+    get_multiple_data
+)
+from remme.settings.helper import _get_setting_value
+LOGGER = logging.getLogger(__name__)
+
+FAMILY_NAME = 'node_account'
+FAMILY_VERSIONS = ['0.1']
+
+
+class NodeAccountHandler(BasicHandler):
+    def __init__(self):
+        super().__init__(FAMILY_NAME, FAMILY_VERSIONS)
+
+    def get_state_processor(self):
+        return {
+            NodeAccountMethod.INITIALIZE_MASTERNODE: {
+                PB_CLASS: NodeAccountInternalTransferPayload,
+                PROCESSOR: self._initialize_masternode,
+                VALIDATOR: NodeAccountInternalTransferPayloadForm,
+            },
+            NodeAccountMethod.CLOSE_MASTERNODE: {
+                PROCESSOR: self._close_masternode,
+                PB_CLASS: CloseMasternodePayload,
+                VALIDATOR: CloseMasternodePayloadForm,
+            },
+        }
+
+    def _initialize_masternode(self, context, node_account_public_key, internal_transfer_payload):
+        node_account_address = self.make_address_from_data(node_account_public_key)
+
+        node_account = get_data(context, NodeAccount, node_account_address)
+
+        if node_account is None:
+            raise InvalidTransaction('Invalid context or address.')
+
+        if node_account.node_state != NodeAccount.NEW:
+            raise InvalidTransaction('Masternode is already opened or closed.')
+
+        if node_account.balance < internal_transfer_payload.value:
+            raise InvalidTransaction('Insufficient amount of tokens on operational account.')
+
+        minimum_stake = int(_get_setting_value(context, SETTINGS_MINIMUM_STAKE))
```

Comment:
https://github.com/Remmeauth/remme-core/pull/233#discussion_r261522347

Reply:
Done. Check for None and integer.